### PR TITLE
FW-5175 join for small screens

### PR DESCRIPTION
**Pull request by gmcauliffe**
### Description of Changes

- Adjustments to styling to optimize join form for mobile and smaller screens
- Adjustments to PrivateSite modal message styling to optimize for mobile
- Replace deprecated isLoading in SIteData
- Reduce mobile menu item size to fit on smaller screens 

### Checklist

- [ ] README / documentation has been updated if needed
- [ ] PR title / commit messages contain Jira ticket number if applicable
- [ ] Tests have been updated / created if applicable

### Reviewers Should Do The Following:

- [ ] Review the changes here
- [ ] Pull the branch and test locally

### Additional Notes

N/A

### CODE REVIEW

File: src/components/NavBar/NavBarPresentationMobile.js
Comment (sahil97) [inline src/components/NavBar/NavBarPresentationMobile.js:27]: is it safe to remove the optional chaining operator here ? I came across a error in the FE code last week while debugging with Cara which was caused due to a missing optional chaining operator, as the code was trying to pass down the site?.XX variable before the site object was loaded in memory, causing the site to crash.

**Review by sahil97 — APPROVED** [review]
Except for the small doubt I have, LGTM !

File: src/components/NavBar/NavBarPresentationMobile.js
Comment (gmcauliffe) [reply src/components/NavBar/NavBarPresentationMobile.js:27]: Good question, Sarah and I came across the same issue last week. The button for the mobile menu, along with most of the Navbar, is now wrapped in a check for site loading so it shouldn't be an issue. I will double check it on Dev once it is merged before passing to QA. Thanks!

**Pull request merged**
